src/contracts/pods/EigenPodManager.sol

```diff
@@ -215,6 +225,36 @@ contract EigenPodManager is
         ownerToPod[podOwner].withdrawRestakedBeaconChainETH(destination, shares);
     }
 
+    /// @notice Called by proving service to fulfill a partial withdrawal proof request
+    function proofServiceCallback(
+        bytes32 blockRoot,
+        uint64 oracleTimestamp,
+        WithdrawalCallbackInfo[] calldata callbackInfo
+    ) external onlyProofService partialWithdrawalProofSwitchOn {
```

Comment:
Let's make this nonReentrant since we're calling an untrusted address in the EigenPod